synapse/metrics/background_process_metrics.py

```diff
@@ -214,10 +213,7 @@ def run():
             except Exception:
                 logger.exception("Background process '%s' threw an exception", desc)
             finally:
-                proc.update_metrics()
-
-                with _bg_metrics_lock:
-                    _background_processes[desc].remove(proc)
+                _background_process_in_flight_count.labels(desc).dec()
```

Comment:
Has the background process completely finished at this point or only when it's paused?

I'm trying to figure out how this differs from `BackgroundProcessLoggingContext.__exit__`, which also decrements `_background_process_in_flight_count`.

Reply:
`BackgroundProcessLoggingContext.__exit__` also gets called here, so you could put it in there, but it currently isn't in there? We discard from the set in `__exit__` and call `update_metrics`, but `_background_process_in_flight_count` is only called in this function

Reply:
Oop, yes I conflated the `discard` line with a `desc` one in my head.

Reply:
Ah, fair :)